pandas/core/internals/blocks.py

```diff
@@ -227,7 +218,13 @@ def _can_hold_na(self) -> bool:
     @final
     @property
     def is_categorical(self) -> bool:
-        return self._holder is Categorical
+        warnings.warn(
```

Comment:
is there something that actually tests this warning?

Reply:
no, its a DeprecationWarning, just for eg pyarrow